### ISSUE TITLE
Populate nullable state for struct members of interfaces and classes

### ISSUE THREAD

**Issue by chsienki** [dotnet/roslyn#34921]
Consider the following:

```csharp
#nullable enable

interface I<T>
{
    T P { get; }
}

class C<T>
{
    internal T F = default!;
}

class Program
{
    static void F1<T>(I<(T, T)> i) where T : class?
    {
        var t = i.P;
        t.Item1.ToString(); // should warn
    }

    static void F2<T>(C<(T, T)> c) where T : class?
    {
        var t = c.F;
        t.Item1.ToString();// should warn
    }
}
```
We don't currently warn for either case.

A second repro is available here: https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQBMQGoA+ABADAAmwEYBuAWACg9CiAWcigYgDsBXAG3YmHbnzmbdelStgDMhAEz4AwpQDelfMvxKVxAGyFa+AGJEAPABUAfAApiYg2aNJ8RgJQn+RB/gDuAC0R8j+EFIA/GrKihQqEfgAbhAI+BD4ALwuAHQA4nAwAKJsYIgQMAD2CGYODJEqMXHASfEpMqwICAIw5RXRsfgAxrXAKQCS8GBEKUaFAMowCACWzADmpW0qAL4hymua2nqSxuaW1rb2Ti5uXj72/kFrYe0dcQnJcCMZ2bn5RSVlaxFV+DXJEHqjWazBgAyGpG+lU6PWSfTGkxm80W31WFGWQA=

**Comment by jcouv** [comment]
This was fixed. Let's add test to confirm.
